x/tools/imports/export.go

```diff
@@ -29,18 +36,39 @@ func GoPath() map[string]*Pkg {
 func exportDirScan(ds map[string]*pkg) map[string]*Pkg {
 	r := make(map[string]*Pkg)
 	for path, pkg := range ds {
-		r[path] = exportPkg(pkg)
+		p, err := exportPkg(pkg)
+		if err != nil {
+			continue
+		}
+
+		r[path] = p
 	}
 	return r
 }
 
 // Pkg represents exported type of pkg.
 type Pkg struct {
 	Dir             string // absolute file path to Pkg directory ("/usr/lib/go/src/net/http")
+	Name            string // package name ("http", "a")
 	ImportPath      string // full Pkg import path ("net/http", "foo/bar/vendor/a/b")
 	ImportPathShort string // vendorless import path ("net/http", "a/b")
 }
 
-func exportPkg(p *pkg) *Pkg {
-	return &Pkg{Dir: p.dir, ImportPath: p.importPath, ImportPathShort: p.importPathShort}
+func exportPkg(p *pkg) (*Pkg, error) {
+	fset := token.NewFileSet()
+	pkgs, err := parser.ParseDir(fset, p.dir, notGoTestFile, parser.PackageClauseOnly)
```

Comment:
Hmmm..... I think this change make gopkgs slow down. Parsing for every packages seems high cost.

Do you have any thought? Do you measure end-to-end speed?

If it's slow, I think we should skip parsing process if the users doesn't need "Name".
Maybe provide "-name" flag instead of "-f" or something.

Reply:
Yes it slower... it reads the directory to get the package name.

Using `-name` will only show the name part. 
Flag `-f` can be used example `gopkgs -f '{{.Name}}:{{.ImportPath}}'` so we can get pair of package name and the import path.

Original version:
```
$ time gopkgs|wc -l
    3386
gopkgs  0.09s user 0.17s system 219% cpu 0.118 total
wc -l  0.00s user 0.00s system 2% cpu 0.118 total
```

Modified version:
```
$ time gopkgs|wc -l
    3381
gopkgs  0.68s user 0.68s system 303% cpu 0.446 total
wc -l  0.00s user 0.00s system 0% cpu 0.445 total
```

Reply:
@haya14busa I think the performance still acceptable.
Every file read it only read the until package name.

Reply:
It's not acceptable. It's 10x slower.
There could be thousands and thousands go files under GOPATH.
IO may occur every time for each files.

You can add "-include-name" flag, or parse files only if "-format" is specified.